irr/src/CGLTFMeshFileLoader.cpp

```diff
@@ -0,0 +1,693 @@
+						? model.bufferViews->at(*accessor.bufferView).byteStride.value_or(accessor.elementSize())
+						: accessor.elementSize());
+
+		for (std::size_t i = 0; i < overriddenCount; ++i) {
+			u32 index;
+			std::visit([&](auto &&acc) { index = acc.get(i); }, indicesAccessor);
+			if (index >= accessor.count)
+				throw std::runtime_error("index out of bounds");
+			vec[index] = valuesAccessor.get(i);
+		}
+		return Accessor<T>(vec, accessor.count);
+	}
+
+	return base;
+}
+
+#define ACCESSOR_TYPES(T, U, V)                                                             \
+	template <>                                                                             \
+	constexpr tiniergltf::Accessor::Type SelfType::Accessor<T>::getType()                   \
+	{                                                                                       \
+		return tiniergltf::Accessor::Type::U;                                               \
+	}                                                                                       \
+	template <>                                                                             \
+	constexpr tiniergltf::Accessor::ComponentType SelfType::Accessor<T>::getComponentType() \
+	{                                                                                       \
+		return tiniergltf::Accessor::ComponentType::V;                                      \
+	}
+
+#define VEC_ACCESSOR_TYPES(T, U, n)                                                                        \
+	template <>                                                                                            \
+	constexpr tiniergltf::Accessor::Type SelfType::Accessor<std::array<T, n>>::getType()                   \
+	{                                                                                                      \
+		return tiniergltf::Accessor::Type::VEC##n;                                                         \
+	}                                                                                                      \
+	template <>                                                                                            \
+	constexpr tiniergltf::Accessor::ComponentType SelfType::Accessor<std::array<T, n>>::getComponentType() \
+	{                                                                                                      \
+		return tiniergltf::Accessor::ComponentType::U;                                                     \
+	}                                                                                                      \
+	template <>                                                                                            \
+	std::array<T, n> SelfType::rawget(const void *ptr)                                                     \
+	{                                                                                                      \
+		const T *tptr = reinterpret_cast<const T *>(ptr);                                                  \
```

Comment:
this can cause problems e.g. on ARM if `T` has higher alignment requirements than the actual alignment of `ptr`

but I'm not even sure in which case this is used.

Reply:
This is also a strict aliasing violation. I would've preferred to have this new stuff PRd to my fork so I could do a cursory review of it first.

Reply:
I fixed this before. This change is a regression.

Reply:
Likely not. There should be a compiler warning on some versions of GCC with a high enough warning level, and that's probably the only indication in _most_ cases.

Reply:
should be fixed in https://github.com/minetest/minetest/pull/14557/commits/fbc53588fba31262da0ce744846d8260bb014180